subsys/net/lib/config/init_clock_sntp.c

```diff
@@ -11,12 +11,30 @@ LOG_MODULE_DECLARE(net_config, CONFIG_NET_CONFIG_LOG_LEVEL);
 #include <zephyr/net/sntp.h>
 #include <zephyr/posix/time.h>
 
+static int sntp_init_helper(struct sntp_time *tm)
+{
+#ifdef CONFIG_NET_CONFIG_SNTP_INIT_SERVER_USE_DHCPV4_OPTION
+	struct net_if *iface = net_if_get_default();
+
+	if (iface->config.dhcpv4.ntp_addr.s_addr != 0) {
+		struct sockaddr_in sntp_addr = {0};
+
+		sntp_addr.sin_family = AF_INET;
+		sntp_addr.sin_addr.s_addr = iface->config.dhcpv4.ntp_addr.s_addr;
+		return sntp_simple_sock((struct sockaddr *)&sntp_addr, sizeof(sntp_addr),
+					CONFIG_NET_CONFIG_SNTP_INIT_TIMEOUT, tm);
+	}
+	LOG_INF("SNTP address not set by DHCPv4, using Kconfig defaults");
+#endif /* NET_CONFIG_SNTP_INIT_SERVER_USE_DHCPV4_OPTION */
+	return sntp_simple(CONFIG_NET_CONFIG_SNTP_INIT_SERVER, CONFIG_NET_CONFIG_SNTP_INIT_TIMEOUT,
```

Comment:
Quite long line that is a bit hard to read, suggest placing the timeout config value to next line.